packages/frontend/src/features/metricsCatalog/components/MetricsTableTopToolbar.tsx

```diff
@@ -252,6 +235,17 @@ export const MetricsTableTopToolbar: FC<Props> = memo(
             FeatureFlags.MetricTrees,
         );
 
+        const isValidMetricsTree =
+            totalResults > 0 && totalResults <= MAX_METRICS_TREE_NODE_COUNT;
+
+        // If the metrics tree is not valid, set the view to list
+        // can happen when changing filters while in tree view
```

Comment:
Good enough for now, but the experience is a little odd. You mess with filters and the view changes without warning. Something to think about.

Reply:
I mean something for all of us to think about 😄. Should we disable filters on the tree? It would be cool if they work, but at the moment itll almost always pop you out of the tree experience to mess with them. Unless you are being very careful to only narrow down.

Reply:
I completely agree, I was going to leave this as is until we have a better experience but this can be annoying, I will try to use the suboptimal state component, hopefully it will work fine here 🙏

Reply:
Much better!